php/php.editor/src/org/netbeans/modules/php/editor/actions/FixUsesPerformer.java

```diff
@@ -631,13 +757,221 @@ private int getOffsetWithoutLeadingWhitespaces(final int startOffset) {
         return result;
     }
 
+    private int getOffsetWithoutFollowingWhitespaces(final int endOffset) {
+        int result = endOffset;
+        baseDocument.readLock();
+        try {
+            TokenSequence<PHPTokenId> ts = LexUtilities.getPHPTokenSequence(baseDocument, endOffset);
+            if (ts != null) {
+                ts.move(endOffset);
+                while (ts.moveNext() && ts.token().id().equals(PHPTokenId.WHITESPACE)) {
+                    result = ts.offset() + ts.token().length();
+                }
+            }
+        } finally {
+            baseDocument.readUnlock();
+        }
+        return result;
+    }
+
+    private class UsesInsertStringHelper {
```

Comment:
Can the class be `static`?

Reply:
It can, if I make editList static or put it directly to the constructor. Could you suggest what the advantages in such case? Sorry for probably silly question, but I am not very good in Java for now. Thanks.

Reply:
The non-static inner class cannot exist without it's outer class. Therefore, it is preferred to make inner classes static, if possible.

Reply:
But in this case it shouldn't exist without outer class because it's using outer class attribute and, although it isn't so critical for short living objects, but the static classes are increasing the memory consumption. I will put editList directly to the helper constructor, but the logic still isn't clear enough for me.

Reply:
I agree with @tmysik .
The member class should be `static` if it does not need access to the enclosing class instance.
If you are interested in it, I suggest trying to read Effective Java(Item 24: Favor static member classes over nonstatic).